### ISSUE TITLE
Troubleshooting Section

### ISSUE THREAD

**Issue by felipeelia** [10up/wp-local-docker-v2#120]
**Is your enhancement related to a problem? Please describe.**

I've been using WP Local Docker as my development env for 3 months now and I had some problems properly configuring it. I think this kind of info would fit nicely into a "Troubleshooting" section or page.

**Describe the solution you'd like**

We could either create a new page in the docs or adding a section to the Installation page or perhaps use GitHub's Wiki. I don't know what would be better.

I'll leave two problems and solutions here, hopefully, these can be a start or at least they be found by those who need it.

**Problem 1: NVM**

Problem: Any 10updocker command won't modify the `hosts` file while using nvm
Description: To modify the `hosts` file we call the 10updocker-hosts command with sudo. NVM will add its node versions to PATH only for the current user, not for root. That said, calling `sudo 10updocker-hosts add something`, for example, will return a `command not found` for those using NVM.
Solution: We can add a symbolic link in `/usr/local/bin` pointing to the needed script. Something like
```
sudo ln -s $(which 10updocker-hosts) /usr/local/bin/10updocker-hosts
```

**Problem 2: dnsmasq on Linux Mint**

Problem: Linux Mint (probably Ubuntu too) uses the systemd-resolved service to resolve DNS queries. That completely bypasses the dnsmasq global container.
Solution: Linux uses the `/etc/resolv.conf` to know which server it should use to resolve addresses. Adding `nameserver 127.0.0.1` to the top of the file fixes the problem.

**Designs**
n/a

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by jmichaelward** [comment]
+1 to this ticket, with the additional note that I'm personally encountering Problem 2 on Pop!_OS, a variant of Ubuntu. I already have a `nameserver 127.0.01` entry in my `/etc/resolv.conf` file, but attempting to create an environment causes the build fail due to port conflicts if the dnsmasq service is running. Disabling it, of course, allows it to create.

Naturally, port conflicts are a concern for Docker, as any running services will cause various container builds to fail. My e-mail provider requires me to run a local server to resolve e-mail, for instance, so the Mailcatcher container also causes issues. Fortunately, in my case, I am able to change the port of that service, but that may not always be possible.

Another troubleshooting issue as it pertains to Linux is the issue of Docker being restricted to root user access by default. [There is official documentation](https://docs-stage.docker.com/engine/install/linux-postinstall/#manage-docker-as-a-non-root-user) which suggests steps for allowing non-root user access, but which include security cautions as well. 10updocker will not properly create the PHP-FPM container if these steps are not first followed on Linux environments.

**Comment by eugene-manuilov** [comment]
The 1st problem will be solved when #123 is merged.

**Comment by tylercherpak** [comment]
Hey all, I think these are resolved in the latest version. Also I encourage anyone to submit PRs for the [F.A.Q section](https://github.com/10up/wp-local-docker-v2/#faq). If you have any questions, let us know. Thanks!
